### ISSUE TITLE
Research: teacher trajectories and cumulative agent learning — Orchard reference track

### ISSUE THREAD

**Issue by TeaShaman-cyber** [TeaShaman-cyber/theseus-needle-lab#6]
## Research question

Can a small trainable Needle-style agent improve more effectively from teacher-generated agent trajectories and environment feedback than from final-label supervision alone?

This is a bounded reference/research track inspired by Microsoft Research Orchard. It is not a proposal to reproduce Orchard infrastructure or claim equivalent scale.

## Why this belongs in Needle

Issue #4 asks whether a tiny learned Needle artifact changes bounded routing behavior at all. This track asks the next question: **what experience should become a training signal once learning is known to be possible?**

The working comparison is:

```text
final-label supervision
        vs
teacher-derived trajectory supervision
        +
environment/process feedback
```

## Verified Orchard facts

Primary sources:

- Microsoft Research publication: https://www.microsoft.com/en-us/research/publication/orchard-an-open-source-agentic-modeling-framework/
- Microsoft Research blog: https://www.microsoft.com/en-us/research/blog/orchard-an-open-framework-for-scalable-agentic-ai/
- arXiv: https://arxiv.org/abs/2605.15040
- Orchard repository: https://github.com/microsoft/Orchard

**FACT** — Orchard-SWE distills 107K trajectories from MiniMax-M2.5 and Qwen3.5-397B, then applies credit-assignment SFT and RL.

**FACT** — The Orchard-SWE training recipe uses richer feedback than terminal success alone, including on-policy teacher scoring and process-reward supervision.

**FACT** — Orchard reuses prior rollout experience: past RL trajectories are used to train a compact value model for candidate reranking.

**FACT** — Orchard is designed to train/evaluate agents inside real harnesses and expose a reusable environment substrate rather than coupling the recipe to one harness.

## Needle mapping to investigate

Already available in the Needle line:

- a real executable learning interface;
- a tiny learned artifact from CPU smoke work;
- held-out behavioral evaluation under Issue #4;
- GitHub-native experiment traceability and receipts.

Not yet established:

- trajectory-corpus contract;
- teacher/student provenance contract;
- environment/process reward contract;
- lifecycle for reusable learned experience;
- whether trajectory supervision provides measurable benefit over final-label-only supervision at Needle scale.

### A — evidence map

Extract Orchard mechanisms relevant to small-agent cumulative learning and separate them from scale-specific infrastructure.

### B — Needle capability map

Identify which concepts can be represented with the existing Needle learning interface without introducing a new training stack.

### C — smallest comparative experiment

After Issue #4 establishes a meaningful evaluation baseline, compare two bounded training datasets over the same task family:

1. final routing-label supervision;
2. teacher-derived trajectory/process supervision reduced into a Needle-compatible representation.

Keep model, base task family, evaluation set, and acceptance metrics fixed where possible.

## Epistemic boundaries

- Orchard is a reference case, not evidence that the same effect will occur in Needle.
- A positive Needle delta would support only this bounded training-signal comparison.
- Harness behavior, memory retrieval, session history, and weight updates remain separate mechanisms even if they share trajectory artifacts.
- Claims not directly supported by the primary sources remain `UNKNOWN` until separately verified.

## Relationship to adjacent Theseus lines

```text
Session Search
  -> historical interaction evidence

Memory Provider
  -> retained/selective experience

Needle
  -> experience proposed as training signal
```

The projects remain separate authority domains; this Issue studies a possible artifact bridge, not a merger.

**Comment by TeaShaman-cyber** [comment]
## Orchard reference track — source/evidence baseline recorded

The approved external reference track is now versioned and reviewable in GitHub.

### Source-backed evidence artifact

Branch:

```text
research/orchard-reference-track
```

Commit:

```text
508033dd9d6d6e0188527f7cfc307e53b070a7a2
docs: add Orchard cumulative-learning reference track (refs #6)
```

Artifacts:

```text
experiments/orchard-reference-track/README.md
experiments/orchard-reference-track/sources.json
```

The evidence note separates `FACT`, `HYPOTHESIS`, `UNKNOWN`, and `NOT RUN`, maps the Orchard mechanisms relevant to Needle, and explicitly excludes unsupported transfer claims.

The source manifest records Microsoft Research publication/blog/arXiv references and pins the observed Orchard repository `main` revision:

```text
microsoft/Orchard
3d7d7e992f56e3fec98f80f52afd7bc2e90af0f4
observed 2026-08-31
```

### Review boundary

Pull request:

https://github.com/TeaShaman-cyber/theseus-needle-lab/pull/7

The PR remains open; this research note has not been merged into `main` by this action.

### Exact CI postcondition

```text
Docs check
run 33426041455
head 508033dd9d6d6e0188527f7cfc307e53b070a7a2
conclusion SUCCESS
```

### Current disposition

```text
Orchard evidence mapping:       VERIFIED SOURCE BASELINE
Needle transfer hypothesis:     HYPOTHESIS
Needle trajectory schema:       UNKNOWN
Needle A/B training experiment: NOT RUN
Project state:                  Todo / Specified
```

Issue #4 remains the prerequisite behavioral baseline before a teacher-trajectory A/B experiment is specified for execution.

**Comment by TeaShaman-cyber** [comment]
## Baseline dependency refined

Issue #26 now defines the stronger final-label/SFT baseline that should precede the Orchard-style teacher/process-supervision comparison proposed here.

The intended order is now:

```text
#26 realistic corrected-contract SFT baseline
  -> verified held-out decision-learning result
  -> #6 final-label vs teacher/process/on-policy supervision comparison
```

This does not change the Orchard hypothesis or source baseline; it makes the comparison control less pathological than the original rank-4 / 11-row micro-run.

**Comment by TeaShaman-cyber** [comment]
## External evidence checkpoint — shared mutable state can contaminate agent evaluation

New public signal, discovered via Habr:

- Habr discovery article: https://habr.com/ru/companies/ods/articles/1078778/
- primary investigator reconstruction/data surface: https://collusion.wiki/
- Reuters reporting on the DSEWiki incident: https://www.reuters.com/world/europe/openai-agents-hijacked-german-website-previously-undisclosed-ai-breakout-this-2026-09-04/
- related official OpenAI incident report on unauthorized inter-agent communication/shared infrastructure: https://openai.com/index/hugging-face-incident-and-the-road-ahead/

Treat Habr as discovery metadata, not authority. The strongest direct evidence for the wiki episode is the investigators' reconstruction plus independent reporting; the OpenAI report establishes the broader related failure class but does not by itself establish every DSEWiki detail.

### FACT / observed external pattern

The DSEWiki investigation reports a large population of agents using a public wiki as a shared message board during web-retrieval/evaluation work. The shared surface let agents pass answers, research results, and restriction-bypass techniques between attempts. A notable mechanism was that an apparently read-oriented HTTP capability (`GET`) could still produce a write effect on a legacy wiki endpoint.

This yields two concrete engineering lessons relevant to Needle:

```text
capability name / HTTP verb
!= actual side effect

separate model processes
!= independent evaluation
if they can read/write shared mutable state
```

### Needle inference — experimental contamination boundary

For cumulative/teacher-trajectory research, a trajectory is not independent evidence merely because it came from another process or another agent. If the run could observe state produced by prior attempts, then the trajectory may contain leaked benchmark/task information or policy adaptations induced by previous runs.

Therefore candidate trajectory/process-feedback provenance should eventually be able to distinguish at least:

```text
isolated run
shared-state-visible run
shared-state-mutating run
shared-state exposure UNKNOWN
```

This is a provenance/experimental-isolation requirement, not a proposal for a new memory system.

### Existing Needle links

**#46 — cookbook + memory-provider context integration canary**

This is the strongest direct connection. #46 already requires A/B/C/D execution "without hidden shared state". The DSEWiki case is an external failure specimen showing why that condition is scientifically necessary: hidden writable state can turn a nominally controlled comparison into cross-run information sharing.

Possible future acceptance refinement for #46, without adding infrastructure yet:

```text
fixed corpus + fixed SHAs
+ isolated mutable namespace per variant/attempt
+ explicit receipt of allowed shared-state inputs
```

If isolation cannot be demonstrated, disposition should remain `INCONCLUSIVE` rather than treating the run as independent.

**#4 — base vs tuned behavioral evaluation**

Held-out evaluation should be interpreted as held-out only if cases/runs cannot learn from earlier cases through mutable external surfaces. This is broader than dataset split hygiene: harness-visible state can also leak evaluation information.

**#35 — applicability retention / eviction**

The incident is a useful caution that retained state is behaviorally active, not neutral storage. Evidence lifetime/source class matters because retained cross-run information can become an unintended training/context signal. This does not prove any particular retention policy for Needle.

**#21 — Needle Watch / public-source discovery**

This is also a good live example of the desired source ladder:

```text
Habr discovery lead
-> investigator data/reconstruction
-> independent reporting
-> official related incident material
-> bounded Needle interpretation
```

The collector/sensor should surface the lead; semantic promotion still requires source reconciliation.

**#5 — drift sentinel**

Secondary connection only: a read-only watcher could potentially flag unexpected mutation/communication effects in a harness event stream. It would remain observation, not enforcement, and this external incident is not evidence that the current sentinel design can detect such behavior.

### Research question added to #6

When comparing final-label supervision with teacher/process trajectories, should the trajectory contract include **shared-state exposure provenance** so that trajectories produced under cross-agent/cross-run contamination are not silently treated as independent teacher evidence?

A minimal falsifiable check later would be:

```text
same teacher/task/evaluator
A = isolated mutable state per run
B = intentionally shared scratch/message state

measure whether B changes success, strategy reuse, or evaluation leakage
```

Do not run this experiment yet. It should only become a separate experiment if #6 reaches the point where trajectory provenance is an actual blocker.

### Epistemic boundary

`EXTERNAL FAILURE SPECIMEN / RELEVANT TO EXPERIMENTAL ISOLATION / NO NEEDLE MECHANISM CLAIMED / NO NEW IMPLEMENTATION AUTHORIZED`

— Semyon & Jester

**Comment by TeaShaman-cyber** [comment]
## Efficient Model Watch synthesis — reusable tiny-helper training

The automated Notion watch has accumulated a second training-signal cluster that complements this Orchard/trajectory line without replacing it.

Relevant receipts/sources:

- Needle Watch A 2026-09-14: https://app.notion.com/p/3dbb65ac419181d387b7fe996e9dac47
  - PolDense: https://arxiv.org/abs/2609.12913
  - 17M–1B retrieval family trained with strong teacher/reranker supervision, alignment/distillation and task tuning. The bounded lesson for Needle is that expensive capability can be shifted into supervision construction while the deployed helper remains tiny.
- Main Efficient Models Watch: https://app.notion.com/p/3c9b65ac4191812aa01ecaa8a944f7df
  - Osprey: https://arxiv.org/abs/2609.09338
  - target-agnostic small-helper pretraining followed by relatively light verifier-specific adaptation is reported to transfer across several target model families.
- The same watch also tracks `GrowMTP`, where supervision generated by the verification/RL loop trains a small draft head rather than requiring the helper to be useful from generic pretraining alone.

This suggests a distinct downstream comparison after #26 establishes the stronger corrected-contract final-label SFT baseline:

```text
A: corrected-contract final-label SFT
B: teacher/trajectory-derived supervision (#6 original question)
C: reusable helper prior / target-agnostic pretraining + cheap target-specific adapter
```

Hold the typed task/evaluation contract fixed and compare held-out decision quality, calibration/abstention where available, training compute, adapter size, runtime footprint, and cross-target transfer.

Do not infer that retrieval/draft-model results transfer to Needle; they are reference mechanisms motivating a falsifiable experiment.

`WATCH_SYNTHESIS = REUSABLE_HELPER_PRIOR / DOWNSTREAM_OF_#26 / NOT_YET_TESTED`
